src/pages/ExamsPage/components/AttemptList.test.jsx

```diff
@@ -12,12 +12,21 @@ jest.unmock('react');
 jest.mock('../hooks', () => ({
   useDeleteExamAttempt: jest.fn(),
   useModifyExamAttempt: jest.fn(),
+  useExamsData: jest.fn(),
 }));
 
 describe('AttemptList', () => {
+  const defaultExamsData = {
```

Comment:
Because I'm now getting the currentExam via the useExamsData hook, I'm needing to mock this fn and create a default exams data in both AttemptList and the ReviewExamAttemptButton. I don't love that I need to have it in the AttemptList since it's just the parent component - wondering if there is a better way to mock this. I see a console error when these tests were failing about error boundaries: https://legacy.reactjs.org/docs/error-boundaries.html so I can look into that too if that's better.

Reply:
You could try mocking out the actual children of `AttemptList` (e.g. `ReviewExamAttemptButton`)? That way, you're only testing the functionality in the `AttemptList` component without rendering the `ReviewExamAttemptButton` and needing to mock out the `useExamsData` hook.

However, I suspect that's not the testing strategy the React Testing Library would recommend, right? If I recall correctly, their approach is to test the application as closely as it would appear to the user. Mocking out children components is not that.

I was reading this [thread](https://github.com/testing-library/react-testing-library/issues/167) from some ideas. It's a little old, though. But, based on the conversation in that thread, I wonder if the tests in `ReviewExamAttemptButton.jsx` should be moved into `AttemptList.test.jsx`, and the `AttemptList.test.jsx` acts as the E2E testing suite for that whole component?

I'm not sure what the right approach is here, but these are just some ideas. Maybe a good topic for standup?

Reply:
Notes:
- In the learning assistant library, we set up a mock redux store. That could also solve the problem potentially. > is this recommended?
- What we were using before worked well for mocking child components. React Testing Library kind of wants you to put together the whole testing state.

Reply:
maybe a simpler path than following the learning MFE, which handles _everything_ (rendering, i18n, redux) is just to have a helper that populates a default state. That's pretty close to what we're already doing on the base component. Wouldn't require any large refactoring to make something like this a default for all tests: https://github.com/edx/frontend-app-exams-dashboard/blob/main/src/pages/ExamsPage/index.test.jsx#L26-L40

Reply:
Cool! Refactored to have a default for all tests in testutils.